src/test/Makefile

```diff
@@ -42,6 +42,9 @@ C_FLAGS = $(COMMON_FLAGS) \
 CXX_FLAGS = $(COMMON_FLAGS) \
 	-std=gnu++11
 
+# Determine the OS
+UNAME := $(shell uname)
+#
```

Comment:
And use:
```
ifneq ($(UNAME), Darwin)
PARAMETER_GROUP = -Wl,-T,$(TEST_DIR)/parameter_group.ld
endif
```
There is no need to pass linker script on darwin.

```
$(CXX) $(CXX_FLAGS) $(PARAMETER_GROUP)  $^ -o $(OBJECT_DIR)/$@
```

@hydra: We can pass linker script in LDFLAGS (should be implemented) / CXX_FLAGS to all targets. Empty section gets discarded by `ld`, so it is safe. And more sections may appear in future, so maybe even drop parameter_group reference (`linker_script.ld`)